docs/source/contributing/python-style-guide.md

```diff
@@ -0,0 +1,492 @@
+# Python Style Guide
+
+This document is a _diff_ - it describes how Numenta's Python guidelines differ from or are unspecified by [PEP-8](http://www.python.org/dev/peps/pep-0008/) and [PEP-257](http://www.python.org/dev/peps/pep-0257/).
+
+It's meant to be succinct.  Generally, we wish to remain close to the best practices of the Python community.
+
+## Python Version
+
+Numenta software requires Python 2.7.
+
+## Pylint
+
+The NuPIC installation should put `pylint` on your path and you should create a symbolic link at `~/.pylintrc` that points to `$NUPIC/py/pylintrc`.  The pylint command enforces many of the style guidelines from this document and PEP8.
+
+## File Layout
+
+### Executables
```

Comment:
We have changed this. Now, executables should be specified in `setup.py` but are otherwise just functions somewhere in the project.